### PR TITLE
Clarify syntax for qualified names of operators in manual

### LINKED ISSUE #19235: quoting Module.operator for method definition
I was trying to define a method for `Base.==` **without** importing it.
```julia
julia> Base.==(::Foo,::Foo) = true
ERROR: syntax: "(::Foo,::Foo)" is not a valid function argument name
julia> Base.:==(::Foo,::Foo) = true
ERROR: syntax: invalid identifier name ":="
julia> Base.(==)(::Foo,::Foo) = true
ERROR: syntax: invalid function name "Base.((==,))"
```
Cf
```julia
julia> Base.:+(::Foo,::Foo) = Foo()
```
Which works fine. The issue is the following:
1. the manual could talk about the syntax for quoting operator names when qualified by a module,
2. if there is a uniform syntax, it should work for `==` too.

Using `v"0.5.0"`.

### PR DESCRIPTION
**Pull request by Tetralux**
~~Fixes~~ Addresses #19235.

### CODE REVIEW

**Comment by H-225** [comment]
Travis failure seems unrelated?

File: doc/manual/modules.rst [hunk 1]
```diff
@@ -234,7 +234,13 @@ Namespace miscellanea
 ---------------------
 
 If a name is qualified (e.g. ``Base.sin``), then it can be accessed even if
-it is not exported. This is often useful when debugging.
+it is not exported. This is often useful when debugging. It can also have
+methods added to it by using the qualified name as the function name.
+However, due to syntactic ambiguities that arise, if you wish to add methods
+to a function in a different module whos name contains only symbols, such as
```
Comment (JeffBezanson) [inline doc/manual/modules.rst]: whos => whose

**Review by JeffBezanson — CHANGES_REQUESTED** [review]

File: doc/manual/modules.rst (hunk 1 above)
Comment (H-225) [inline doc/manual/modules.rst]: Done.

**Comment by musm** [comment]
@ScottPJones  mentioned that someone should post this here, but that he couldn't since he is banned

Line 240: whos should be whose  (already fixed)
Line 242/243:
+If the operator is more than one character in length you must surround it in
+brackets, such as: Base.:(==).
The parsing bug isn't whether not the operator is more than one character in length,
it seems to be only related to operators starting with =,
_and I think we normally use American English in the documentation, so brackets should be parentheses_

File: doc/manual/modules.rst [hunk 2]
```diff
@@ -234,7 +234,13 @@ Namespace miscellanea
...
 
 If a name is qualified (e.g. ``Base.sin``), then it can be accessed even if
-it is not exported. This is often useful when debugging.
+it is not exported. This is often useful when debugging. It can also have
+methods added to it by using the qualified name as the function name.
+However, due to syntactic ambiguities that arise, if you wish to add methods
+to a function in a different module whose name contains only symbols, such as
+an operator, ``Base.+`` for example, you must use ``Base.:+`` to refer to it.
+If the operator is more than one character in length you must surround it in
+brackets, such as: ``Base.:(==)``.
```
Comment (musm) [inline doc/manual/modules.rst]: parentheses

**Pull request merged**